# see #546416
# Thanks to Bill Allombert <ballombe@debian.org>
# fixed by Andi Barth
diff -ur netpbm-free-10.0.old/ppm/ppmtompeg/jpeg.c netpbm-free-10.0/ppm/ppmtompeg/jpeg.c
--- netpbm-free-10.0.old/ppm/ppmtompeg/jpeg.c	2003-08-12 20:23:03.000000000 +0200
+++ netpbm-free-10.0/ppm/ppmtompeg/jpeg.c	2009-09-14 10:47:26.000000000 +0200
@@ -435,6 +435,9 @@
   cinfo.want_raw_output = TRUE;
 #else
   cinfo.raw_data_out = TRUE;
+#if JPEG_LIB_VERSION >= 70
+  cinfo.do_fancy_upsampling = FALSE;
+#endif
 #endif
   cinfo.out_color_space = JCS_YCbCr;
   
@@ -480,6 +483,8 @@
   /* Make an 8-row-high sample array that will go away when done with image */
 #ifdef JPEG4
   buffer_height = 8;  /* could be 2, 4,8 rows high */
+#elif JPEG_LIB_VERSION >= 70
+  buffer_height = cinfo.max_v_samp_factor * cinfo.min_DCT_v_scaled_size;
 #else
   buffer_height = cinfo.max_v_samp_factor * cinfo.min_DCT_scaled_size;
 #endif
